src/OrchardCore.Modules/OrchardCore.Users/Workflows/Activities/SelectUsersInRoleTask.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.Services;
+using OrchardCore.Workflows.Abstractions.Models;
+using OrchardCore.Workflows.Activities;
+using OrchardCore.Workflows.Models;
+using OrchardCore.Workflows.Services;
+
+namespace OrchardCore.Users.Workflows.Activities
+{
+    public class SelectUsersInRoleTask : TaskActivity
+    {
+        private readonly UserManager<IUser> _userManager;
+        private readonly IUserService _userService;
+        private readonly IWorkflowExpressionEvaluator _expressionEvaluator;
+        protected readonly IStringLocalizer S;
+
+        public SelectUsersInRoleTask(UserManager<IUser> userManager, IUserService userService, IWorkflowExpressionEvaluator expressionvaluator, IStringLocalizer<SelectUsersInRoleTask> localizer)
+        {
+            _userManager = userManager;
+            _userService = userService;
+            _expressionEvaluator = expressionvaluator;
+            S = localizer;
+        }
+
+        public override string Name => nameof(SelectUsersInRoleTask);
+
+        public override LocalizedString DisplayText => S["Select Users in Role Task"];
+
+        public override LocalizedString Category => S["User"];
+
+        public WorkflowExpression<string> PropertyName
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> RoleName
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(S["Done"], S["Empty"], S["Failed"]);
+        }
+
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var propName = await _expressionEvaluator.EvaluateAsync(PropertyName, workflowContext, null);
+            var roleName = await _expressionEvaluator.EvaluateAsync(RoleName, workflowContext, null);
+
+            if (!string.IsNullOrEmpty(propName) && !string.IsNullOrEmpty(roleName))
```

Comment:
I think all you need here is something like this to get all users in a given role

```
var expression = await _expressionEvaluator.EvaluateAsync(RoleNames, workflowContext, null);
var roleNames = JsonSerializer.Deserialize<string[]>(expression ?? string.Empty);

var users = new List<IUser>();

foreach (var roleName in roleNames)
{
    users.AddRange(await _userManager.GetUsersInRoleAsync(roleName));
}
```

Reply:
Not sure why you need this. If you are removing a role, not sure why do you need the email. Maybe you need another task that would locate the user? Like write the user object into the workflow output, then email task would read the output and extract the email.